### PR TITLE
LPS-151368 fill the languageId attribute from the default of the structure if current language is not available.

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPS-151368 if the language id is not filled, the language is taken from the httpServletRequest and the default language is not filled so the default language is not saved, only the language from the request is sent but the document is not localized so the values on DDMFieldAttribute are empty

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa 

there is no comment on my page.
maybe it failed to be sent with the review and it's "pendiente"?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp [hunk 1]
```diff
@@ -462,6 +462,7 @@ renderResponse.setTitle(headerTitle);
 												containerId='<%= liferayPortletResponse.getNamespace() + "dataEngineLayoutRenderer" + ddmStructure.getStructureId() %>'
 												dataDefinitionId="<%= ddmStructure.getStructureId() %>"
 												dataRecordValues="<%= ddmFormValuesToMapConverter.convert(ddmFormValues, DDMStructureLocalServiceUtil.getStructure(ddmStructure.getStructureId())) %>"
+												languageId="<%= LocaleUtil.toLanguageId(themeDisplay.getSiteDefaultLocale()) %>"
```
[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:465]: Oh!

I was so centered on this rotten tree that I forget the forest!!

But I tested our way of change the language (the flag) and the documents works fine.

On the case of the videos (with the changes) the fields always are: 
defaultLanguageId":"en_US","editingLanguageId":"en_US",

And if not the "editingLanguageId":"es_ES" (or the language of the user)

On the case of the other DT localizables the values are sent and the value on "localizedValueEdited":{"ar_SA":true,"en_US":true,"es_ES":true,"ca_ES":true,"nl_NL":true},"editingLanguageId":"ca_ES",


but I think that maybe delegating the value of the languageId to the displayContext is more accurate.

Are you ok to create another method on the BaseDLEditFileEntryDisplayContext and an expecific method on DLVideoExternalShortcutDLEditFileEntryDisplayContext ??

[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:465]: Yes, I'm ok with that. Can you also try to do it in a generic way? I was thinking of something like (in pseudo java):
```
public String getDocumentTypeLanguageId() {
    String languageId = themeDisplay.getLanguageId(); // Current user language.

    if (!ddmStructure.getAvailableLanguageIds().contains(languageId)) {
        return ddmStructure.getDefaultLanguageId();
    }
    return languageId;
}
```

[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:465]: I was thinking more on:

On DLEditFileEntryDisplayContext: 
```
public String getDocumentTypeLanguageId() {
    return LanguageUtil.getLanguageId(PortalUtil.getLocale(httpServletRequest));
}
```
This is like the value that the code on DataLayoutRendererTag put when the value of languageId is not present

And on

On DLVideoExternalShortcutDLEditFileEntryDisplayContext
```
@Override
public String getDocumentTypeLanguageId() {
    return LocaleUtil.toLanguageId(themeDisplay.getSiteDefaultLocale());
}
```

To be the only "special" case by now that uses a different language id

[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:465]: and on the 
`
<liferay-data-engine:data-layout-renderer`

put`

languageId="<%= dlEditFileEntryDisplayContext. getDocumentTypeLanguageId() %>"`

[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:465]: @AliciaGarciaGarcia  ?

[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:465]: I'm on it today, the previous days I was with other task

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**